metal/v1/version.go

```diff
@@ -0,0 +1,13 @@
+package v1
+
+import "strings"
+
+// Version of the metal-go package.  The version is passed in as a parameter
+// to the generator, and we parse it out of the default UserAgent at runtime
+var Version = "(unused)"
+
+func init() {
```

Comment:
init runs at startup in every package that will include metal-go. Do we already have another init?

If we can avoid doing this, that would be prefered.

Could we perhaps take advantage of custom templates to create our own version.go template file that uses the packageVersion variable? https://openapi-generator.tech/docs/customization/#custom-generator-and-template

Reply:
https://github.com/influxdata/influx-cli/ looks to be doing some custom templating and similar processes with OpenAPI Tools Generator against on a Go project.

Reply:
We could use a custom template to change how known entities are generated (APIs, client, models, configuration), but we can't introduce a new entity via custom templates; configuration is probably the most sensible place to inject the version, but I'm not sure I see the value in introducing custom templates just to expose the package version.  How would you feel about removing `Version` entirely?  As far as I can tell, the only reason it exists is so that [`cluster-api-provider-packet` can include the SDK version in its own version info](https://github.com/kubernetes-sigs/cluster-api-provider-packet/pull/527/files#diff-f6bf0ae0d21c9408c6624174da11dafb369ce306f2cdcea704285d8acc87e19eR61), and we could find a different approach over there (build args, use the user-agent instead, move the user-agent parsing to that codebase, etc.) instead of adding code here.

Reply:
That sounds good - we don't need the version baked in since we have the User-Agent. I assume we can still prefix the UA as demonstrated here: https://github.com/equinix/terraform-provider-equinix/pull/291/files#diff-5d56dbc70ed6632fd0523ea2cb89e3afc4a82621b68c6d4010cabfd5cafb2304R215

We could also take this approach if needed: https://github.com/packethost/packngo/blob/master/version.go
(my argument against `init()` is in no way nullified by this example: 🤦)

Reply:
It _looks like_ the MetalSDKVersion is not used: https://github.com/search?q=repo%3Akubernetes-sigs%2Fcluster-api-provider-packet%20MetalSdkVersion&type=code